### DISCUSSION TITLE
Component Not Sowing Data after Update

### DISCUSSION THREAD

**Discussion by abbasmashaddy72** [livewire/livewire#5139] [1. Help]
I am Using a From Proton media Component From

Which working fine in all my project I am having issue with just only this one case Where I am unable to show the date in view whereas when I am doing dd I am able to get the data in loop that I have Created

View:
`<x-custom-form>
    @wire('debounce.200ms')
        <x-form-select name="patient_id" label="Select Patient" :options="Helper::getKeyValues('Patient', 'patient', 'id')" placeholder="Please Select" />

        <x-form-select name="doctor_id" label="Select Doctor" :options="Helper::getKeyValues('Doctor', 'doctor', 'id')" placeholder="Please Select" />

        @if (!empty($doctor_id))
            @foreach ($appointment_dates as $key => $value)
                <x-form-label label="{{ $value->day }}" />
                @foreach (Helper::getTimeSlots($value->from, $value->to, $value->appointment_duration) as $item)
                    <x-form-radio name="appointment_date" label="{{ $item }}" />
                @endforeach
            @endforeach
        @endif

        <x-form-select name="referral_id" label="Select Referral Doctor" :options="Helper::getKeyValues('Referral', 'doctor', 'id')" placeholder="Please Select" />
    @endwire
</x-custom-form>`

Component:
`<?php

namespace App\Http\Livewire\Form;

use App\Models\DoctorSchedule;
use Livewire\Component;

class AppointmentCES extends Component
{
    // Model Values
    public $doctor_id, $patient_id, $referral_id, $appointment_date;

    //Custom Values
    public $appointment_dates;

    public function updatedDoctorId($doctor)
    {
        $this->appointment_dates = DoctorSchedule::where('doctor_id', $doctor)->get();
        $this->emitSelf('refreshComponent');
    }

    public function render()
    {
        return view('livewire.form.appointment-c-e-s');
    }
}
`

Data from Debug Bar:
`2lm
array:5 [
  "data" => array:5 [
    "doctor_id" => "2"
    "patient_id" => null
    "referral_id" => null
    "appointment_date" => null
    "appointment_dates" => Illuminate\Database\Eloquent\Collection {#1882
      #items: array:3 [
        0 => App\Models\DoctorSchedule {#1922
          #connection: "mysql"
          #table: "doctor_schedules"
          #primaryKey: "id"
          #keyType: "int"
          +incrementing: true
          #with: []
          #withCount: []
          +preventsLazyLoading: false
          #perPage: 15
          +exists: true
          +wasRecentlyCreated: false
          #escapeWhenCastingToString: false
          #attributes: array:8 [
            "id" => 211
            "doctor_id" => 2
            "day" => "Monday"
            "from" => "12:00:00"
            "to" => "24:00:00"
            "appointment_duration" => 5
            "created_at" => "2022-07-29 05:35:25"
            "updated_at" => "2022-07-29 05:35:25"
          ]
          #original: array:8 [
            "id" => 211
            "doctor_id" => 2
            "day" => "Monday"
            "from" => "12:00:00"
            "to" => "24:00:00"
            "appointment_duration" => 5
            "created_at" => "2022-07-29 05:35:25"
            "updated_at" => "2022-07-29 05:35:25"
          ]
          #changes: []
          #casts: []
          #classCastCache: []
          #attributeCastCache: []
          #dates: []
          #dateFormat: null
          #appends: []
          #dispatchesEvents: []
          #observables: []
          #relations: []
          #touches: []
          +timestamps: true
          #hidden: []
          #visible: []
          #fillable: array:4 [
            0 => "day"
            1 => "from"
            2 => "to"
            3 => "appointment_duration"
          ]
          #guarded: array:1 [
            0 => "*"
          ]
        }
        1 => App\Models\DoctorSchedule {#1878
          #connection: "mysql"
          #table: "doctor_schedules"
          #primaryKey: "id"
          #keyType: "int"
          +incrementing: true
          #with: []
          #withCount: []
          +preventsLazyLoading: false
          #perPage: 15
          +exists: true
          +wasRecentlyCreated: false
          #escapeWhenCastingToString: false
          #attributes: array:8 [
            "id" => 336
            "doctor_id" => 2
            "day" => "Wednesday"
            "from" => "12:00:00"
            "to" => "24:00:00"
            "appointment_duration" => 15
            "created_at" => "2022-07-29 05:35:25"
            "updated_at" => "2022-07-29 05:35:25"
          ]
          #original: array:8 [
            "id" => 336
            "doctor_id" => 2
            "day" => "Wednesday"
            "from" => "12:00:00"
            "to" => "24:00:00"
            "appointment_duration" => 15
            "created_at" => "2022-07-29 05:35:25"
            "updated_at" => "2022-07-29 05:35:25"
          ]
          #changes: []
          #casts: []
          #classCastCache: []
          #attributeCastCache: []
          #dates: []
          #dateFormat: null
          #appends: []
          #dispatchesEvents: []
          #observables: []
          #relations: []
          #touches: []
          +timestamps: true
          #hidden: []
          #visible: []
          #fillable: array:4 [
            0 => "day"
            1 => "from"
            2 => "to"
            3 => "appointment_duration"
          ]
          #guarded: array:1 [
            0 => "*"
          ]
        }
        2 => App\Models\DoctorSchedule {#1712
          #connection: "mysql"
          #table: "doctor_schedules"
          #primaryKey: "id"
          #keyType: "int"
          +incrementing: true
          #with: []
          #withCount: []
          +preventsLazyLoading: false
          #perPage: 15
          +exists: true
          +wasRecentlyCreated: false
          #escapeWhenCastingToString: false
          #attributes: array:8 [
            "id" => 381
            "doctor_id" => 2
            "day" => "Thursday"
            "from" => "12:00:00"
            "to" => "24:00:00"
            "appointment_duration" => 15
            "created_at" => "2022-07-29 05:35:25"
            "updated_at" => "2022-07-29 05:35:25"
          ]
          #original: array:8 [
            "id" => 381
            "doctor_id" => 2
            "day" => "Thursday"
            "from" => "12:00:00"
            "to" => "24:00:00"
            "appointment_duration" => 15
            "created_at" => "2022-07-29 05:35:25"
            "updated_at" => "2022-07-29 05:35:25"
          ]
          #changes: []
          #casts: []
          #classCastCache: []
          #attributeCastCache: []
          #dates: []
          #dateFormat: null
          #appends: []
          #dispatchesEvents: []
          #observables: []
          #relations: []
          #touches: []
          +timestamps: true
          #hidden: []
          #visible: []
          #fillable: array:4 [
            0 => "day"
            1 => "from"
            2 => "to"
            3 => "appointment_duration"
          ]
          #guarded: array:1 [
            0 => "*"
          ]
        }
      ]
      #escapeWhenCastingToString: false
    }
  ]
  "name" => "form.appointment-c-e-s"
  "view" => "livewire.form.appointment-c-e-s"
  "component" => "App\Http\Livewire\Form\AppointmentCES"
  "id" => "FCqivWTWgrWf3eGjC2lm"
]`

With dd Screenshot:
![Screenshot_20220729_122322](https://user-images.githubusercontent.com/55267488/181700952-06fa36f3-cf02-4950-a3ee-054cb7cc8665.png)

Without dd Screenshot:
![Screenshot_20220729_122343](https://user-images.githubusercontent.com/55267488/181701030-7666d781-9039-4a4e-9489-d99ab9168176.png)

Please Help Me Out
Thanks in advance...

**Comment by joshhanley** [comment]
@abbasmashaddy72 it seems like you are trying to use an `@wire` and `@endwire` tags, but they aren't offical Livewire tags. Where did you get them from?

**Reply by abbasmashaddy72** [reply]
I am using this Package: https://github.com/protonemedia/laravel-form-components

**Reply by abbasmashaddy72** [reply]
It renders the same livewire Syntax that we write normally

**Comment by joshhanley** [comment]
@abbasmashaddy72 ah ok, I've not seen that package before!

It's possible you might be running into a dom diffing issue https://laravel-livewire.com/docs/2.x/troubleshooting#dom-diffing-issues
One thing is when you are using conditional elements like `@if` or `@foreach` it's best to wrap the conditional in a `div` which makes it easier for Livewire to process.

So putting a div around `@if (!empty($doctor_id))` like:

```blade
    <x-form-select name="doctor_id" label="Select Doctor" :options="Helper::getKeyValues('Doctor', 'doctor', 'id')" placeholder="Please Select" />

    <div>
        @if (!empty($doctor_id))
            @foreach ($appointment_dates as $key => $value)
                <x-form-label label="{{ $value->day }}" />
                @foreach (Helper::getTimeSlots($value->from, $value->to, $value->appointment_duration) as $item)
                    <x-form-radio name="appointment_date" label="{{ $item }}" />
                @endforeach
            @endforeach
        @endif
    </div>

    <x-form-select name="referral_id" label="Select Referral Doctor" :options="Helper::getKeyValues('Referral', 'doctor', 'id')" placeholder="Please Select" />
```

Hope this helps!

**Reply by abbasmashaddy72** [reply]
It is not working

You can help me out by checking my code on GitHub: https://github.com/abbasmashaddy72/umdaa_main

You can also install it as the same Laravel installation method also already created dummy migration data

Step 1: composer install
Step 2: npm i && npm run dev(using latest vite)
Step 3: php artisan migrate:fresh --seed (This will Some Time as there are many tables)
Step 4: php artisan serve

Your URL will be http://127.0.0.1:8000/appointment/create

Livewire View: form/appointment-c-e-s
Livewire Component: Form/AppointmentCES

Thanks in advance for helping out

**Reply by abbasmashaddy72** [reply]
Hi

@joshhanley

**Reply by joshhanley** [reply]
@abbasmashaddy72 I'm currently not in a position to pull your repo, but if I get the chance, I will pull your repo have a look. 

But we generally prefer not to pull peoples main repos as it takes a lot more time (especially if things take a lot of time to seed) than just reviewing code in a discussion and responding 🙂

We prefer if you could reproduces the issue with as little code as possible and share that code or create a small example repo with only that code so we can investigate it. If you can that I'd be more able to have a look sooner. 🙂

Thanks!

**Reply by abbasmashaddy72** [reply]
Okay, I will make the issue and share the repo, thanks @joshhanley

**Reply by joshhanley** [reply]
Thanks!
